extensions/src/org/pentaho/platform/web/http/filters/PentahoPathDecodingFilter.java

```diff
@@ -58,5 +124,40 @@ public String getRequestURI() {
       final String requestURI = super.getRequestURI();
       return requestURI.replaceAll( "%255C", "%5C" ).replaceAll( "%252F", "%2F" );
     }
+
+    @Override
+    @Deprecated // too many conversions
+    public Map<String, String[]> getParameterMap() {
+      if ( parameterMap == null ) {
+        return super.getParameterMap();
+      }
+      return getNativeMap();
+    }
+
+    @Override
+    public String getParameter( String name ) {
+      if ( parameterMap == null ) {
+        return super.getParameter( name );
+      }
+      return parameterMap.getFirst( name );
+    }
+
+    @Override
+    public String[] getParameterValues( String name ) {
+      if ( parameterMap == null ) {
+        return super.getParameterValues( name );
+      }
+      List<String> values = parameterMap.get( name );
+      return values.toArray( new String[values.size()] );
+    }
+
+    @Override
+    public Enumeration<String> getParameterNames() {
+      if ( parameterMap == null ) {
+        return super.getParameterNames();
+      }
+      return new Vector<String>( parameterMap.keySet() ).elements();
```

Comment:
@mbatchelor I didn't found how better to create the Enumeration from Set. But now i found the variant "Iterators.asEnumeration( parameterMap.keySet().iterator() );". Is it ok?